tortoisebeacon/beacon_calc.go

```diff
@@ -35,6 +35,8 @@ func (tb *TortoiseBeacon) calcBeacon(epoch types.EpochID, coinflip bool) error {
 	tb.beacons[epoch] = beacon
 	tb.beaconsMu.Unlock()
 
+	tb.cleanupVotes(epoch)
```

Comment:
what happens if we exit with error in this method or even before calcBeacon is called?

Reply:
The call needs to be deferred in a parent function